src/wallet/coinselection.cpp

```diff
@@ -335,12 +337,18 @@ util::Result<SelectionResult> KnapsackSolver(std::vector<OutputGroup>& groups, c
         for (const auto& group : applicable_groups) {
             result.AddInput(group);
         }
+        // We may still fail if max is low
+        if (result.GetWeight() > max_weight) return ErrorMaxWeightExceeded();
```

Comment:
Would this be giving up too early? We could still find a different solution further down the function, no?

Reply:
yeah I was mostly hot-patching here, making sure it never returns too big(because it did very often)

Reply:
should be mitigated?